frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectKeyAvailabilityAssuranceAction.java

```diff
@@ -0,0 +1,28 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAwareAction;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyProject;
+
+public abstract class BaseProjectKeyAvailabilityAssuranceAction extends DumbAwareAction {
+  @Override
+  @UIEffect
+  public void update(AnActionEvent anActionEvent) {
+    super.update(anActionEvent);
+
+    if (this instanceof IExpectsKeyProject) {
+      IExpectsKeyProject thisAction = (IExpectsKeyProject) this;
+      var projectOption = thisAction.tryGetProject(anActionEvent);
+      if (projectOption.isEmpty()) {
+        return;
+      }
+    }
+
+    onUpdate(anActionEvent);
+  }
+
+  @UIEffect
+  protected abstract void onUpdate(AnActionEvent anActionEvent);
```

Comment:
Actually, to reduce the risk of someone forgetting to call `super.onUpdate`, I'd define `onUpdate` in `BaseGitMacheteRepositoryReadyAction` instead, and mark `update` as final there. This way nobody will need to remember to call `super.onUpdate` since it's gonna be always called from `BaseGitMacheteRepositoryReadyAction#update`.

Reply:
The only problem will be then that classes that extend `BaseProjectKeyAvailabilityAssuranceAction` but not `BaseGitMacheteRepositoryReadyAction`... but IIRC there's currently just one (?) such class anyway, so we can call `super.update` explicitly from that class.

Reply:
Ok, but what about actions that are not `BaseGitMacheteRepositoryReadyAction`? What about `BaseProjectKeyAvailabilityAssuranceAction`? Should only `update` method be there and then actions that derived from `BaseProjectKeyAvailabilityAssuranceAction` should `@Override` `update` method and call `super#update`?

Reply:
As agreed via Slack we leave this as is for consistency: in every action we override `onUpdate` method and call `super#onUpdate`

Reply:
+ grep check for call `super`